dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java

```diff
@@ -0,0 +1,133 @@
+        String dockstoreURl = consumer.getConsumerConfiguration().getString("dockstoreURL");
+        String zenodoToken = consumer.getConsumerConfiguration().getString("zenodoToken");
+        String zenodoURL = consumer.getConsumerConfiguration().getString("zenodoURL");
+
+
+        AmazonSQS sqs = AmazonSQSClientBuilder.standard().withRegion(Regions.US_EAST_1).build();
+        LOG.info("Receiving messages from MyFifoQueue.fifo");
+        ReceiveMessageRequest receiveMessageRequest = new ReceiveMessageRequest(sqsURL);
+        // there's no indefinite wait
+        receiveMessageRequest.setMessageAttributeNames(Lists.newArrayList(".*"));
+        receiveMessageRequest.setWaitTimeSeconds(MAX_WAITTIME_SECONDS);
+        receiveMessageRequest.setMaxNumberOfMessages(1);
+        Gson gson = new Gson();
+
+        DOIHandler doiHandler = new DOIHandler(dockstoreURl, dockstoreToken, zenodoURL, zenodoToken);
+        Map<String, MessageHandler> messageHandlers = new HashMap<>();
+        messageHandlers.put(doiHandler.messageTypeHandled(), doiHandler);
+
+        do {
+            List<Message> messages = sqs.receiveMessage(receiveMessageRequest).getMessages();
+            for (Message message : messages) {
+                LOG.debug("  Message");
+                LOG.debug("    MessageId:     " + message.getMessageId());
+                LOG.debug("    ReceiptHandle: " + message.getReceiptHandle());
+                LOG.debug("    MD5OfBody:     " + message.getMD5OfBody());
+                LOG.debug("    Body:          " + message.getBody());
+                for (Entry<String, String> entry : message.getAttributes().entrySet()) {
+                    LOG.debug("  Attribute");
+                    LOG.debug("    Name:  " + entry.getKey());
+                    LOG.debug("    Value: " + entry.getValue());
+                }
+
+                // in reality, get the workflow here and create a zenodo entry for it
+                MessageHandler messageHandler = messageHandlers.get(message.getMessageAttributes().get("type").getStringValue());
+                if (messageHandler != null) {
+                    BasicMessage basicMessage = (BasicMessage)gson.fromJson(message.getBody(), messageHandler.messageClassHandled());
+                    boolean handled = messageHandler.handleMessage(basicMessage);
+
+                    String messageReceiptHandle = messages.get(0).getReceiptHandle();
+                    if (handled) {
+                        // Delete the message
+                        LOG.info("Deleting the message");
+                        sqs.deleteMessage(new DeleteMessageRequest(sqsURL, messageReceiptHandle));
+                    } else {
+                        // requeue the message
+                        sqs.changeMessageVisibility(sqsURL, messageReceiptHandle, 0);
+                    }
+                }
+
+            }
+        } while (true);
```

Comment:
How often does this loop execute?

Reply:
It is intended to consume messages ad infinitum till shutdown

Reply:
I was more wondering which situations should we use some sort of polling rate vs just using infinite loop.

Reply:
Depends, this prototype uses the maximimum AWS setting of 20 seconds to timeout per request. 
We can tinker to see if a different rate is more appropriate